core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -512,6 +514,14 @@ public List<Snapshot> snapshots() {
     return snapshots;
   }
 
+  public void setKekCache(Map<String, KeyEncryptionKey> kekCache) {
```

Comment:
I'll change this to `withKekCache`, making the implementation similar to [`withUUID`](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/TableMetadata.java#L554-L556) (via new Builder).